apps/designer/app/designer/features/props-panel/control.tsx

```diff
@@ -117,109 +160,186 @@ const BooleanControl = ({
   />
 );
 
-const RangeControl = ({
-  value,
-  defaultValue,
-  onChange,
-  min,
-  max,
-  step,
-}: RangeControlProps) => (
-  <Flex direction="column" gap={1}>
-    <Slider
-      value={value}
-      defaultValue={defaultValue}
-      onValueChange={(values) => {
-        onChange(values[0]);
-      }}
-      min={min}
-      max={max}
-      step={step}
-    />
-    <Flex direction="row" justify="between">
-      <Text>{min}</Text>
-      <Text>{max}</Text>
-    </Flex>
-  </Flex>
-);
+type ImageControlProps = {
+  asset: Asset | null; // no image
+  onChange: (asset: Asset) => void;
+};
 
-const NotImplemented = () => <div />;
+const ImageControl = ({ asset, onChange }: ImageControlProps) => {
+  // TODO: implement image control
+
+  // const { assets } = useAssets("image");
+
+  // const selectedAsset = assets.find((asset) => asset.id === value.value);
+  // console.info(assets);
 
-type PrimitiveControlProps = BaseControlProps & {
-  type: "array" | "boolean" | "date" | "number" | "object" | "text";
+  return (
+    <ValuePickerPopover
+      title="Images"
+      content={
+        <ImageManager
+          onChange={(asset) => {
+            // setValue(asset.id);
+            // eslint-disable-next-line no-console
+            onChange(asset);
+          }}
+        />
+      }
+    >
+      <TextField defaultValue={asset?.name ?? " - "} />
+    </ValuePickerPopover>
+  );
 };
 
-type ColorControlProps = BaseControlProps & {
-  type: "color";
-  presetColors?: Array<string>;
+const NotImplemented = () => <div />;
+
+const includes = <T extends string>(arr: readonly T[], v: string): v is T => {
+  return arr.includes(v as never);
 };
 
-type FileControlProps = BaseControlProps & {
-  type: "file";
-  accept: string;
+const isStringArray = (arr: unknown): arr is Array<string> => {
+  return Array.isArray(arr) && arr.every((v) => typeof v === "string");
 };
 
-type RangeControlProps = BaseControlProps<number> & {
-  type: "range";
-  min: number;
-  max: number;
-  step: number;
+const assertUnreachable = (_arg: never) => {
+  throw new Error("Didn't expect to get here");
```

Comment:
Would be good to include `_arg` in the message. Just in case :)

Reply:
Its typescript hack, see `_arg: never` so any call to this function with any non `never` argument would cause ts error. `throw new Error` just instead of comment. IMO no need to check typescript in runtime adding any logic on never arg or output it

Reply:
I mean what if a user reports a bug saying they're getting a error "Didn't expect to get here"? It would be better for us if user comes with something like "Didn't expect to get here. Got: something".

Reply:
Moved msg out of scope. Yep found a case that it can be executed in runtime - for example, if a developer will forget to set `return` in the code below, it would be no ts error.